### PR TITLE
DDCNLS-2061 Fix too restrictive guard in close correspondence address

### PR DESCRIPTION
**Pull request by AndrewArkless**

### CODE REVIEW

File: test/controllers/address/ClosePostalAddressControllerSpec.scala [hunk 1]
```diff
@@ -232,6 +233,42 @@ class ClosePostalAddressControllerSpec extends AddressBaseSpec {
       verify(controller.editAddressLockRepository, times(0)).insert(meq(nino.withoutSuffix), meq(PostalAddrType))
     }
 
+    "render the thank you page upon successful submission of closing the correspondence address and only a lock on the primary address" in new LocalSetup {
+      override def currentRequest[A]: Request[A] = FakeRequest("POST", "/test").asInstanceOf[Request[A]]
+      override def getEditedAddressIndicators: List[AddressJourneyTTLModel] =
+        List(AddressJourneyTTLModel("SomeNino", EditPrimaryAddress(BSONDateTime(0))))
+      val result = controller.confirmSubmit(FakeRequest())
+
+      status(result) shouldBe OK
```
Comment (oc-ste) [inline test/controllers/address/ClosePostalAddressControllerSpec.scala:242]: How do we know if this is redirecting to the thank you page?

File: test/controllers/address/ClosePostalAddressControllerSpec.scala [hunk 2]
```diff
@@ -232,6 +233,42 @@ class ClosePostalAddressControllerSpec extends AddressBaseSpec {
...
+      verify(controller.editAddressLockRepository, times(1)).insert(meq(nino.withoutSuffix), meq(PostalAddrType))
+    }
+
+    "render the thank you page upon successful submission of closing the correspondence address and only a lock on the sole address" in new LocalSetup {
+      override def currentRequest[A]: Request[A] = FakeRequest("POST", "/test").asInstanceOf[Request[A]]
+      override def getEditedAddressIndicators: List[AddressJourneyTTLModel] =
+        List(AddressJourneyTTLModel("SomeNino", EditSoleAddress(BSONDateTime(0))))
+      val result = controller.confirmSubmit(FakeRequest())
+
+      status(result) shouldBe OK
```
Comment (oc-ste) [inline test/controllers/address/ClosePostalAddressControllerSpec.scala:260]: Same comment as above

File: test/controllers/address/ClosePostalAddressControllerSpec.scala (hunk 1 above)
Comment (AndrewArkless) [reply test/controllers/address/ClosePostalAddressControllerSpec.scala:253]: expanded the test

File: test/controllers/address/ClosePostalAddressControllerSpec.scala (hunk 2 above)
Comment (AndrewArkless) [reply test/controllers/address/ClosePostalAddressControllerSpec.scala:272]: expanded the test

**Pull request merged**